File: code/mathematical-algorithms/tower_of_hanoi/tower_of_hanoi.ml
(* Part of Cosmos by OpenGenus Foundation *)

let tower_of_hanoi n =
	let rec move n from dest other =
		if n <= 0 then () else (
		move (n-1) from other dest;
		print_endline (from ^ " -> " ^ dest);
		move (n-1) other dest from)
	in
	move n "L" "R" "C"
;;

tower_of_hanoi 5;;

File: code/unclassified/tower_of_hanoi/tower_of_hanoi.py
/*
	towerofhanoi.py
	by AP (@theashwin)
	
	Path of Cosmos by OpenGenus Foundation
*/

def moveTower(height,fromPole, toPole, withPole):
    if height >= 1:
        print( "    "*(3-height), "moveTower:", height, fromPole, toPole )
        moveTower(height-1,fromPole,withPole,toPole)
        moveDisk(fromPole,toPole,height)
        moveTower(height-1,withPole,toPole,fromPole)
    #print(withPole)

def moveDisk(fp,tp,height):
    print("    "*(4-height), "moving disk", "~"*(height), "from",fp,"to",tp)


moveTower(3,"A","B","C")
